native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/utils/ClientSampleSyntaxTreeUtils.java

```diff
@@ -0,0 +1,430 @@
+        }
+    }
+
+    private static ExpressionStatementNode getStreamReceiveValueExpressionStatement(Method method) {
+        ExpressionNode checkExpressionNode = getCheckExpressionNode(getRemoteMethodCallActionNode(
+                getSimpleNameReferenceNode(CONST_STREAMING_CLIENT), "receive" + getMethodType(method.getOutputType())));
+        return getCallStatementNode(checkExpressionNode);
+    }
+
+    private static VariableDeclarationNode getUnaryCallDeclarationNode(Method method, String filename) {
+        TypedBindingPatternNode bindingPatternNode = getTypedBindingPatternNode(
+                getBuiltinSimpleNameReferenceNode(method.getOutputPackageType(filename) + method.getOutputType() + " "),
+                getCaptureBindingPatternNode(CONST_RESPONSE));
+        ExpressionNode node = getCheckExpressionNode(getRemoteMethodCallActionNode(
+                getBuiltinSimpleNameReferenceNode(CONST_ENDPOINT), method.getMethodName(),
+                method.getInputType() != null ? CONST_REQUEST : ""));
+        VariableDeclaration unaryCallVariable = new VariableDeclaration(bindingPatternNode, node);
+        return unaryCallVariable.getVariableDeclarationNode();
+    }
+
+    private static ExpressionStatementNode getUnaryCallExpressionStatement(Method method) {
+        ExpressionNode node = getCheckExpressionNode(getRemoteMethodCallActionNode(
+                getBuiltinSimpleNameReferenceNode(CONST_ENDPOINT), method.getMethodName(),
+                method.getInputType() != null ? CONST_REQUEST : ""));
+        return getCallStatementNode(node);
+    }
+
+    private static VariableDeclarationNode getClientStreamingCallDeclarationNode(Method method) {
+        TypedBindingPatternNode bindingPatternNode = getTypedBindingPatternNode(
+                getBuiltinSimpleNameReferenceNode(
+                        capitalize(method.getMethodName()) + STREAMING_CLIENT + " "),
+                getCaptureBindingPatternNode(CONST_STREAMING_CLIENT));
+        ExpressionNode node = getCheckExpressionNode(getRemoteMethodCallActionNode(
+                getBuiltinSimpleNameReferenceNode(CONST_ENDPOINT), method.getMethodName()));
+        VariableDeclaration streamingCallVariable = new VariableDeclaration(bindingPatternNode, node);
+        return streamingCallVariable.getVariableDeclarationNode();
+    }
+
+    private static ExpressionStatementNode getForEachExpressionNode(Method method, String filename) {
+        FunctionSignatureNode functionSignatureNode = getFunctionSignatureNode(method, filename);
+        AnonymousFunctionExpressionNode expressionNode = getAnonymousFunctionExpressionNode(functionSignatureNode);
+        MethodCallExpressionNode methodCallExpressionNode = getForEachMethodCall(expressionNode);
+        return getCallStatementNode(getCheckExpressionNode(methodCallExpressionNode));
+    }
+
+    private static FunctionSignatureNode getFunctionSignatureNode(Method method, String filename) {
```

Comment:
Can't we re-use the `Function` class[[1]](https://github.com/ballerina-platform/module-ballerina-grpc/blob/d269ad779461978fe548d2817a8103cc8b181a12/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/components/Function.java#L71)? IINM, here we just need to create the `parameters` list using `method` and `filename` is it?

[1] https://github.com/ballerina-platform/module-ballerina-grpc/blob/d269ad779461978fe548d2817a8103cc8b181a12/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/components/Function.java#L71

Reply:
Here what i needed was to create an anonymous function.
For a case like,
```
response.forEach(function (string s) {});
```
I think we cannot use the `Function` api since `functionName` is mandatory there

Reply:
On the other hand I can use `getFunctionSignature` in `Function` without using `getFunctionDefinitionNode`

Reply:
We can create another `Function` constructor which doesn't require a function name. Then change `getFunctionSignature`[[1]](https://github.com/ballerina-platform/module-ballerina-grpc/blob/d269ad779461978fe548d2817a8103cc8b181a12/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/components/Function.java#L91) to `public` and re-use.

[1] https://github.com/ballerina-platform/module-ballerina-grpc/blob/d269ad779461978fe548d2817a8103cc8b181a12/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/components/Function.java#L91

Reply:
Updated in https://github.com/ballerina-platform/module-ballerina-grpc/pull/952/commits/10fa6fdebc07f49e1c8cf191abb3d8b7515638ab